tests/test_dfa.py

```diff
@@ -838,72 +697,74 @@ def test_isfinite_finite(self) -> None:
 
     def test_isfinite_empty(self) -> None:
         """Should test if an empty DFA is finite"""
-        # This DFA has no reachable final states and
-        # therefore is finite.
-        dfa = DFA(
-            states={"q0", "q1", "q2", "q3"},
-            input_symbols={"0", "1"},
-            transitions={
-                "q0": {"0": "q0", "1": "q1"},
-                "q1": {"0": "q1", "1": "q2"},
-                "q2": {"0": "q0", "1": "q1"},
-                "q3": {"0": "q2", "1": "q1"},
-            },
-            initial_state="q0",
-            final_states={"q3"},
-        )
-        self.assertTrue(dfa.isfinite())
+
+        self.assertTrue(self.no_reachable_final_dfa.isfinite())
 
     def test_isfinite_universe(self) -> None:
         # This DFA accepts all binary strings and
         # therefore is infinite.
         dfa = DFA.universal_language({"0", "1"})
         self.assertFalse(dfa.isfinite())
 
-    def test_set_laws(self) -> None:
+    @params(
+        *get_permutation_tuples(
+            DFA.from_substring(set("01"), "1111"),
+            partial_dfa,
+            no_consecutive_11_dfa,
+            at_least_four_ones,
+            zero_or_one_1_dfa,
+            no_reachable_final_dfa,
+        )
+    )
+    def test_set_laws(self, dfa1: DFA, dfa2: DFA) -> None:
         """Tests many set laws that are true for all sets"""
         # This DFA accepts all words which contain at least four
         # occurrences of 1
         input_symbols = {"0", "1"}
-        contains_1111 = DFA.from_substring(input_symbols, "1111")
+
+        dfa1 = dfa1.to_complete()
+        dfa2 = dfa2.to_partial()
+        # dfa1 = DFA.from_substring(input_symbols, "1111")
         # This DFA accepts all words which do not contain two
         # consecutive occurrences of 1
-        avoids_11 = DFA.from_substring(input_symbols, "11").complement(minify=False)
+        # dfa2 = DFA.from_substring(input_symbols, "11").complement(minify=False)
+
+        # DFAs have to be not equal for test to work
+        self.assertNotEqual(dfa1, dfa2)
+
         # This DFA accepts all binary strings
         universal = DFA.universal_language(input_symbols)
         # This DFA represents the empty language
         empty = DFA.empty_language(input_symbols)
         # De Morgan's laws
-        self.assertEqual(~(contains_1111 | avoids_11), ~contains_1111 & ~avoids_11)
-        self.assertEqual(~(contains_1111 & avoids_11), ~contains_1111 | ~avoids_11)
+        self.assertEqual(~(dfa1 | dfa2), ~dfa1 & ~dfa2)
```

Comment:
I think that here I prefer the original names... Makes it more understandable

Reply:
The original names are no longer descriptive. If you take a look at the parametrization of this function, it now tries all pairs of distinct DFAs from the parameters list instead of just two example DFAs.

This is actually the main test case checking for valid interop with complete and partial DFAs now.

Reply:
@eliotwrobson I think the new names are fine, and agree with @eliotwrobson's reasoning for something more descriptive. Closing this thread.